apps/prairielearn/src/models/enrollment.ts

```diff
@@ -509,37 +626,41 @@ export async function setEnrollmentStatus({
 /**
  * Deletes an enrollment.
  */
-export async function deleteEnrollmentById({
-  enrollment_id,
-  agent_user_id,
-  action_detail,
-  agent_authn_user_id,
+export async function deleteEnrollment({
+  enrollment,
+  actionDetail,
+  authzData,
+  requestedRole,
 }: {
-  enrollment_id: string;
-  agent_user_id: string | null;
-  action_detail: SupportedActionsForTable<'enrollments'>;
-  agent_authn_user_id: string | null;
+  enrollment: Enrollment;
+  actionDetail: SupportedActionsForTable<'enrollments'>;
+  authzData: RawAuthzData | DangerousSystemAuthzData;
+  requestedRole: CourseInstanceRole;
 }): Promise<Enrollment> {
+  assertHasRole(authzData, requestedRole, ['Student Data Editor']);
+
   return await runInTransactionAsync(async () => {
-    const oldEnrollment = await selectAndLockEnrollmentById(enrollment_id);
+    const oldEnrollment = await _selectAndLockEnrollment(enrollment.id);
```

Comment:
In `setEnrollmentStatus` you do not use the return value of `_selectAndLockEnrollment`, but here you do. Can we consistently do it one way or the other?